pkgs/development/python-modules/opencv-python/default.nix

```diff
@@ -0,0 +1,17 @@
+{ lib, buildPythonPackage, opencv4, isHeadless ? false, isContrib ? false }:
+
+#TODO: check whether opencv is built with contrib extensions and is not truly headless
+buildPythonPackage {
+  inherit (opencv4) version meta;
+  pname = "opencv${lib.optionalString isContrib "-contrib"}-python${lib.optionalString isHeadless "-headless"}";
+  src = ./.;
```

Comment:
@jonringer I can't seem to make this work as it give the error
```
ERROR: You must give at least one requirement to install (see "pip help install")
```
how can I make this work?

Reply:
You certainly don't want to add a fake package into the `python-modules.nix`. The `python3Packages.opencv4` provides all the same modules, just the package name is different. You really want to

- Either force `pip` to stop looking for the wrong package name (maybe just `pip install --no-dependencies`?)
- Package the actual `opencv-python` wheel from pypi instead of the fake derivation. I'd rather avoid this

Reply:
> Either force `pip` to stop looking for the wrong package name (maybe just `pip install --no-dependencies`?)

I want to avoid `--no-dependencies` as it avoid checking *all* dependencies.

> Package the actual `opencv-python` wheel from pypi instead of the fake derivation. I'd rather avoid this

Frankly this would be better, as it should "simply" be a question of symlinking the correct libraries.

Reply:
`--no-deps` is already used e.g. for source-based opencv when packaging a wheel; I wouldn't treat setuptools as the ultimate source of truth about dependencies anyway, and if there is a testsuite you run later, I'd think it's tolerable to skip pip's dependency check

Reply:
Ok added the flag, now it errors with 
```
; nix build .#python310Packages.fake-background-webcam
def-matcher: command not found
warning: Git tree '/home/me/src/nixpkgs' is dirty
warning: binary cache 'http://127.0.0.1:12304' is for Nix stores with prefix 'Nix::Store::getStoreDir', not '/nix/store'
error: builder for '/nix/store/09vg9km9r4cy4s7lq4sn5xgyk0q88dlg-python3.10-opencv-contrib-python-4.5.4.drv' failed with exit code 1;
       last 10 log lines:
       > source root is opencv-python
       > setting SOURCE_DATE_EPOCH to timestamp 315619200 of file opencv-python/setup.cfg
       > patching sources
       > default.nix  opencv-python.py  pyproject.toml  setup.cfg
       > configuring
       > no configure script, doing nothing
       > installing
       > Executing pipInstallPhase
       > /build/opencv-python/dist /build/opencv-python
       > ERROR: You must give at least one requirement to install (see "pip help install")
       For full logs, run 'nix log /nix/store/09vg9km9r4cy4s7lq4sn5xgyk0q88dlg-python3.10-opencv-contrib-python-4.5.4.drv'.
error: 1 dependencies of derivation '/nix/store/xng0dq225s1x2am1jks5x2c3p0rf29sj-fake-background-webcam-unstable-2021-11-28.drv' failed to build
```

Reply:
Hmm, I don't know what `def-matcher` is. Do you?

Reply:
only something from my shell,nothing relevant at hand